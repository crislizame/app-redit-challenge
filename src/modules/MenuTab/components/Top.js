import React, { Component } from 'react';
import {View, TouchableOpacity, Appearance} from 'react-native'
import { VStack, FlatList, Text, HStack, Image, Center } from "native-base";
import { FontAwesome } from '@expo/vector-icons';
import * as Linking from 'expo-linking';
import {heightPercentageToDP as hp, widthPercentageToDP as wp} from 'react-native-responsive-screen';
import Constants from 'expo-constants';
import { useNavigation } from '@react-navigation/native';
import { WebView } from 'react-native-webview';
import moment from 'moment';

class Top extends Component {

    constructor(props) {
        super(props);
        this.state = {
            schemeColor: Appearance.getColorScheme(),
            data: props.route.params.mockData ? props.route.params.mockData : [],
            type: props.route.params.type,
            isLoading: true
        };
    }  
    _isMounted = false;

    async componentDidMount() {
        this._isMounted = true;
        await this.getData();
    }
    async getData(){
        this.setState({
            isLoading: true
        })
        await fetch(`https://api.reddit.com/r/pics/${this.state.type}.json`, {
            method: 'GET',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json'
            }
        }).then((response) => response.json()).then(async (res) => {

            this.setState({
                data: res.data.children,
                isLoading: false
            });
        })
    }
    async componentWillUnmount() {
        this._isMounted =false;
    }
    render() {
        
        return (
            <VStack testID="top-screen" style={{backgroundColor: this.state.schemeColor === 'light' ? "white" : "black", flex:1, paddingVertical: hp(2), paddingHorizontal: wp(1) }}>
            <FlatList
                testID="flatlist"
                progressViewOffset={5}
                onRefresh={async ()=>{
                    await this.getData()
                }}
                refreshing={this.state.isLoading}
                data={this.state.data}
                renderItem={({ item }) => (
                    <TouchableOpacity activeOpacity={0.8} style={{width:"100%"}} onPress={()=>{
                        this.props.navigation.navigate("WebViewPage",{
                            title:item.data.title,
                            url:`https://reddit.com${item.data.permalink}`
                        })
                    }} >
                        <HStack px={1} py={2} rounded="md" my={2} bg="coolGray.600">
                        <VStack px={2} style={{
                            width:"30%",
                            justifyContent:"center"
                        }}>
                            <Image
                            source={ !item.data.thumbnail || item.data.thumbnail == "nsfw" ? require('../../../../assets/icon.png') : {
                                uri: item.data.thumbnail,
                            }}
                            fallbackSource={require('../../../../assets/icon.png')}
                            alt="Image Reddit"
                            size={wp(20)}
                            style={{
                                resizeMode:"cover",
                                maxHeight: hp(15)
                            }}
                            />
                            
                        
                        </VStack>
                        <VStack style={{justifyContent:"space-between", width:"68%"}}>
                        <Text color="white" style={{textAlign:"right", fontSize:wp(3)}}>{moment.unix(item.data.created).fromNow()}</Text>
                        <Text color="white" style={{fontWeight:"bold", fontSize:wp(4)}}>{item.data.title}</Text>
                        <HStack style={{justifyContent:"space-between"}}>
                            <Center><Text  color="white" style={{fontSize:wp(3)}}>{((item.data.author).length > 12) ? (((item.data.author).substring(0,12-3)) + '...') : item.data.author}</Text></Center>
                            <Center py={1}><Text color="white" style={{fontSize:wp(3)}}><FontAwesome name="star" /> {item.data.score}</Text></Center>
                            <Center><Text color="white" style={{fontSize:wp(3)}}><FontAwesome name="comment" /> {item.data.num_comments}</Text></Center>
                        </HStack>
                        </VStack>
                    </HStack>
                    </TouchableOpacity>
                    
                )}
                keyExtractor={(item, index) => index.toString()}
                />
            </VStack>
        );
    }
}

export default Top
