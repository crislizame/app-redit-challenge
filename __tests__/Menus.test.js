import React from 'react';
import { render } from '@testing-library/react-native'
import MockedNavigator from "../mocks/MockedNavigator";
import Top from '../src/modules/MenuTab/components/Top';
import { NativeBaseProvider, extendTheme } from "native-base";
const wrapper = ({children}) => (
    <NativeBaseProvider
        initialWindowMetrics={{
        frame: {x: 0, y: 0, width: 0, height: 0},
        insets: {top: 0, left: 0, right: 0, bottom: 0},
    }}>
        {children}
    </NativeBaseProvider>
);
const dataFetch = 
    {
        "data": {
            "children": [
                {
                    "kind": "t3",
                    "data": {
                        "title": "Broke my family's poverty cycle - my wife and I just bought our first home!",
                        "link_flair_richtext": [],
                        "subreddit_name_prefixed": "r/pics",
                        "score": 93899,
                        "thumbnail": "https://b.thumbs.redditmedia.com/T2csfDDvwLY6cPxX_g4lTbr9O2OTS3LsTUTGkS3oAeE.jpg",
                        "created": 1627572227.0,
                        "num_comments": 2556,
                        "author": "PrattlesnakeEsquire",
                        "permalink": "/r/pics/comments/otzlkk/broke_my_familys_poverty_cycle_my_wife_and_i_just/"
                    }
                },
            ]
        }
    }

describe('Testing For Menus Top Tab Bar Buttons', ()=>{
    beforeEach(()=>{
        global.fetch = jest.fn(()=> Promise.resolve({
            json: ()=> Promise.resolve(dataFetch)
        }));

    })
    it("Check Top Tab View and test data", ()=>{
        const TopView = render(
            <MockedNavigator component={Top} params={{
                type:"top",
                mockData: dataFetch
            }} />,{wrapper});
        expect(TopView.toJSON()).toMatchSnapshot();
        expect(TopView.getByTestId("top-screen")).toBeDefined();
        expect(TopView.getByTestId('flatlist').props.data.length).not.toEqual(0)
    });
    it("Check Hot Tab View and test data", ()=>{
        const HotView = render(
            <MockedNavigator component={Top} params={{
                type:"hot",
                mockData: dataFetch
            }} />,{wrapper});
        expect(HotView.toJSON()).toMatchSnapshot();
        expect(HotView.getByTestId("top-screen")).toBeDefined();
        expect(HotView.getByTestId('flatlist').props.data.length).not.toEqual(0)
    });
    it("Check New Tab View and test data", ()=>{
        const NewView = render(
            <MockedNavigator component={Top} params={{
                type:"new",
                mockData: dataFetch
            }} />,{wrapper});
        expect(NewView.toJSON()).toMatchSnapshot();
        expect(NewView.getByTestId("top-screen")).toBeDefined();
        expect(NewView.getByTestId('flatlist').props.data.length).not.toEqual(0)
    });

}); 