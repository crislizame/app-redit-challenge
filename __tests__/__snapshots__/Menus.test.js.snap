// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing For Menus Top Tab Bar Buttons Check Hot Tab View and test data 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 44,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 0,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {},
                          Object {
                            "backgroundColor": "white",
                            "flex": 1,
                            "paddingHorizontal": 7.5,
                            "paddingVertical": 26.5,
                          },
                        ]
                      }
                      testID="top-screen"
                    >
                      <RCTScrollView
                        data={
                          Object {
                            "data": Object {
                              "children": Array [
                                Object {
                                  "data": Object {
                                    "author": "PrattlesnakeEsquire",
                                    "created": 1627572227,
                                    "link_flair_richtext": Array [],
                                    "num_comments": 2556,
                                    "permalink": "/r/pics/comments/otzlkk/broke_my_familys_poverty_cycle_my_wife_and_i_just/",
                                    "score": 93899,
                                    "subreddit_name_prefixed": "r/pics",
                                    "thumbnail": "https://b.thumbs.redditmedia.com/T2csfDDvwLY6cPxX_g4lTbr9O2OTS3LsTUTGkS3oAeE.jpg",
                                    "title": "Broke my family's poverty cycle - my wife and I just bought our first home!",
                                  },
                                  "kind": "t3",
                                },
                              ],
                            },
                          }
                        }
                        disableVirtualization={false}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        horizontal={false}
                        initialNumToRender={10}
                        keyExtractor={[Function]}
                        maxToRenderPerBatch={10}
                        onContentSizeChange={[Function]}
                        onEndReachedThreshold={2}
                        onLayout={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onRefresh={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        progressViewOffset={5}
                        refreshControl={
                          <RefreshControlMock
                            onRefresh={[Function]}
                            progressViewOffset={5}
                            refreshing={true}
                          />
                        }
                        refreshing={true}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={Array []}
                        style={
                          Array [
                            Object {},
                          ]
                        }
                        testID="flatlist"
                        updateCellsBatchingPeriod={50}
                        viewabilityConfigCallbackPairs={Array []}
                        windowSize={21}
                      >
                        <RCTRefreshControl />
                        <View />
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Testing For Menus Top Tab Bar Buttons Check New Tab View and test data 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 44,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 0,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {},
                          Object {
                            "backgroundColor": "white",
                            "flex": 1,
                            "paddingHorizontal": 7.5,
                            "paddingVertical": 26.5,
                          },
                        ]
                      }
                      testID="top-screen"
                    >
                      <RCTScrollView
                        data={
                          Object {
                            "data": Object {
                              "children": Array [
                                Object {
                                  "data": Object {
                                    "author": "PrattlesnakeEsquire",
                                    "created": 1627572227,
                                    "link_flair_richtext": Array [],
                                    "num_comments": 2556,
                                    "permalink": "/r/pics/comments/otzlkk/broke_my_familys_poverty_cycle_my_wife_and_i_just/",
                                    "score": 93899,
                                    "subreddit_name_prefixed": "r/pics",
                                    "thumbnail": "https://b.thumbs.redditmedia.com/T2csfDDvwLY6cPxX_g4lTbr9O2OTS3LsTUTGkS3oAeE.jpg",
                                    "title": "Broke my family's poverty cycle - my wife and I just bought our first home!",
                                  },
                                  "kind": "t3",
                                },
                              ],
                            },
                          }
                        }
                        disableVirtualization={false}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        horizontal={false}
                        initialNumToRender={10}
                        keyExtractor={[Function]}
                        maxToRenderPerBatch={10}
                        onContentSizeChange={[Function]}
                        onEndReachedThreshold={2}
                        onLayout={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onRefresh={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        progressViewOffset={5}
                        refreshControl={
                          <RefreshControlMock
                            onRefresh={[Function]}
                            progressViewOffset={5}
                            refreshing={true}
                          />
                        }
                        refreshing={true}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={Array []}
                        style={
                          Array [
                            Object {},
                          ]
                        }
                        testID="flatlist"
                        updateCellsBatchingPeriod={50}
                        viewabilityConfigCallbackPairs={Array []}
                        windowSize={21}
                      >
                        <RCTRefreshControl />
                        <View />
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Testing For Menus Top Tab Bar Buttons Check Top Tab View and test data 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 44,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 0,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {},
                          Object {
                            "backgroundColor": "white",
                            "flex": 1,
                            "paddingHorizontal": 7.5,
                            "paddingVertical": 26.5,
                          },
                        ]
                      }
                      testID="top-screen"
                    >
                      <RCTScrollView
                        data={
                          Object {
                            "data": Object {
                              "children": Array [
                                Object {
                                  "data": Object {
                                    "author": "PrattlesnakeEsquire",
                                    "created": 1627572227,
                                    "link_flair_richtext": Array [],
                                    "num_comments": 2556,
                                    "permalink": "/r/pics/comments/otzlkk/broke_my_familys_poverty_cycle_my_wife_and_i_just/",
                                    "score": 93899,
                                    "subreddit_name_prefixed": "r/pics",
                                    "thumbnail": "https://b.thumbs.redditmedia.com/T2csfDDvwLY6cPxX_g4lTbr9O2OTS3LsTUTGkS3oAeE.jpg",
                                    "title": "Broke my family's poverty cycle - my wife and I just bought our first home!",
                                  },
                                  "kind": "t3",
                                },
                              ],
                            },
                          }
                        }
                        disableVirtualization={false}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        horizontal={false}
                        initialNumToRender={10}
                        keyExtractor={[Function]}
                        maxToRenderPerBatch={10}
                        onContentSizeChange={[Function]}
                        onEndReachedThreshold={2}
                        onLayout={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onRefresh={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        progressViewOffset={5}
                        refreshControl={
                          <RefreshControlMock
                            onRefresh={[Function]}
                            progressViewOffset={5}
                            refreshing={true}
                          />
                        }
                        refreshing={true}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={Array []}
                        style={
                          Array [
                            Object {},
                          ]
                        }
                        testID="flatlist"
                        updateCellsBatchingPeriod={50}
                        viewabilityConfigCallbackPairs={Array []}
                        windowSize={21}
                      >
                        <RCTRefreshControl />
                        <View />
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </View>
</RNCSafeAreaProvider>
`;
